physics/body_test.cpp

```diff
@@ -318,6 +343,73 @@ TEST_F(BodyTest, AllFrames) {
   TestRotatingBody<serialization::Frame::TestTag, serialization::Frame::TO>();
 }
 
+// Check that the rotation of the Earth gives the right solar noon.
+TEST_F(BodyTest, SolarNoon) {
+  struct SurfaceFrame;
+  SolarSystem<ICRS> solar_system_j2000(
+      SOLUTION_DIR / "astronomy" / "sol_gravity_model.proto.txt",
+      SOLUTION_DIR / "astronomy" /
+          "sol_initial_state_jd_2451545_000000000.proto.txt");
+  auto const ephemeris = solar_system_j2000.MakeEphemeris(
+      /*fitting_tolerance=*/5 * Milli(Metre),
+      Ephemeris<ICRS>::FixedStepParameters(
+          SymmetricLinearMultistepIntegrator<QuinlanTremaine1990Order12,
+                                             Position<ICRS>>(),
+          /*step=*/10 * Minute));
+  ephemeris->Prolong("2010-10-01T12:00:00"_UTC);
+
+  auto const earth = solar_system_j2000.rotating_body(*ephemeris, "Earth");
+  auto const sun = solar_system_j2000.rotating_body(*ephemeris, "Sun");
+
+  SphericalCoordinates<double> greenwich;
+  greenwich.radius = 1;
+  greenwich.latitude = 51.4826 * Degree;
+  greenwich.longitude = -0.0077 * Degree;
+  SphericalCoordinates<double> istanbul;
+  istanbul.radius = 1;
+  istanbul.latitude = 41.0082 * Degree;
+  istanbul.longitude = 28.9784 * Degree;
+
+  Vector<double, SurfaceFrame> location;
+
+  auto solar_noon = [earth, &ephemeris, &location, sun](Instant const& t) {
+    Bivector<double, SurfaceFrame> const z({0, 0, 1});
+
+    auto const earth_trajectory = ephemeris->trajectory(earth);
+    auto const sun_trajectory = ephemeris->trajectory(sun);
+    auto const from_surface_frame = earth->FromSurfaceFrame<SurfaceFrame>(t);
+    auto const earth_centre = earth_trajectory->EvaluatePosition(t);
+    auto const sun_centre = sun_trajectory->EvaluatePosition(t);
+    auto const earth_sun = sun_centre - earth_centre;
+    return OrientedAngleBetween(
+        earth_sun, from_surface_frame(location), from_surface_frame(z));
+  };
+
+  location = Vector<double, SurfaceFrame>(greenwich.ToCartesian());
+  auto solar_noon_greenwich = Bisect(solar_noon,
+                                     "2000-01-02T08:00:00"_UTC,
+                                     "2000-01-02T16:00:00"_UTC);
+  EXPECT_THAT(solar_noon_greenwich - "2000-01-02T12:00:00"_UTC,
+              IsNear(4 * Minute, 1.01));
+  solar_noon_greenwich = Bisect(solar_noon,
+                                "2010-09-30T08:00:00"_UTC,
+                                "2010-09-30T16:00:00"_UTC);
+  EXPECT_THAT(solar_noon_greenwich - "2010-09-30T11:00:00"_UTC,
```

Comment:
I think it would be more readable to test `solar_noon_greenwich - "2010-09-30T11:51:00"_UTC` etc.

Reply:
Done.